roles/openshift_health_checker/openshift_checks/efk_stack_check.py

```diff
@@ -0,0 +1,270 @@
+        status['fluentd'] = self.check_fluentd(task_vars)
+        status['kibana'] = self.check_kibana(task_vars)
+
+        return {"changed": True, "msg": status}
+
+    def oc(self, method, args, task_vars):
+        args = {
+            "namespace": self.namespace,
+            "config_file": self.config_file,
+            "method": method,
+            "method_args": args,
+        }
+        result = self.module_executor("ocutil", args, task_vars)
+        return result.get("ansible_facts", {}).get("result", "")
+
+    def get_pods(self, task_vars):
+        """ Get all pods and filter them in one pass """
+        pods = self.oc("get_pods", None, task_vars)
+
+        for pod in pods['items']:
+            if 'component' in pod['metadata']['labels']:
+                # Get ES pods
+                if pod['metadata']['labels']['component'] == 'es':
+                    self.es_pods.append(pod)
+                elif pod['metadata']['labels']['component'] == 'fluentd':
+                    self.fluentd_pods.append(pod)
+
+    def check_elasticsearch(self, task_vars):
+        """ Various checks for elasticsearch """
+        es_status = {}
+        es_status['single_master'] = None
+        es_master_name = None
+        es_status['pods'] = {}
+        for pod in self.es_pods:
+            pod_dc = pod['metadata']['labels']['deploymentconfig']
+            pod_name = pod['metadata']['name']
+            es_status['pods'][pod_dc] = {}
+
+            es_health_check = self.check_elasticsearch_cluster_health(pod_name, task_vars)
+            es_status['pods'][pod_dc]['elasticsearch_health'] = es_health_check
+            es_status['pods'][pod_dc]['disk'] = self.check_elasticsearch_diskspace(pod_name, task_vars)
+
+
+            # Compare the master across all ES nodes to see if we have split brain
+            curl_cmd = "{} 'https://localhost:9200/_cat/master'".format(self.es_curl)
+            es_master = "exec -ti {} -- {}".format(pod_name, curl_cmd)
+            master_name = self.oc("run_user_cmd", es_master, task_vars).split(' ')[1]
+
+            if es_status['single_master'] is None:
+                es_status['single_master'] = 1
+                es_master_name = master_name
+            elif es_master_name != master_name:
+                es_status['single_master'] = 0
+
+        # fix for 3.4 logging where es_master_name is getting set to an ip.
+        # so we set a try check around incase it fails just so it keeps working for 3.3
+        for pod in self.es_pods:
+            try:
+                if pod['status']['podIP'] == es_master_name:
+                    es_master_name = pod['metadata']['name']
+            except:
+                continue
+
+        # get cluster nodes
+        curl_cmd = "{} 'https://localhost:9200/_nodes'".format(self.es_curl)
+        node_cmd = "exec -ti {} -- {}".format(es_master_name, curl_cmd)
+
+        if not es_master_name:
```

Comment:
Why was this check added here?

It looks as if the for loop on line 93 is supposed to fill this field, since it is needed by line 116.  Can we move it before line 116?